why-polars-for-data-cleaning/index.qmd

```diff
@@ -0,0 +1,202 @@
+:::
+
+-   Can be installed and used in Python.
+-   Is designed and built to be fast for processing data.
+-   Has minimal dependencies, so can be installed quickly.
+-   Is flexible to reading different data storage formats (e.g. SQL
+    databases, Parquet, CSV).
+-   Can write to different storage formats (as with above).
+-   We need a tool that can clean data and is either written in Python
+    or has a Python package available.
+
+## Considered options
+
+::: content-hidden
+List and describe some of the options, as well as some of the benefits
+and drawbacks for each option.
+:::
+
+While there are a number of tools available to use for data cleaning,
+given our requirements, these are the options we've considered:
+
+-   [DuckDB](https://www.duckdb.org/)
+-   [Pandas](https://pandas.pydata.org/)
+-   [Polars](https://pola.rs/)
+
+### DuckDB
+
+DuckDB is an in-process online analytical process (OLAP) database tool
+that has API extensions in many languages including Python. It is a very
+young tool, version 1.0.0 was released in 2024, although the first
+version was released in 2019. The developers have set it up as a
+not-for-profit foundation in the Netherlands.
+
+::: columns
+::: column
+#### Benefits
+
+-   Although a very young project it is already in use by a number of
+    large companies and has a large community of users.
+    -   The code is open source, and the way it has been set up is
+        promising for it to remain that way.
+-   It is very fast, and can handle large data sets.
+-   It is designed for data cleaning using SQL syntax.
+-   Extensive documentation with user guide sections embedded.
+:::
+
+::: column
+#### Drawbacks
+
+-   It is not designed exclusively to be used as a data cleaning tool,
+    this is more of an add-on on the way to storing data.
+-   To use it to its full potential, users need to have some knowledge
+    of SQL.
+:::
+:::
+
+### Polars
+
+The Polars project started in 2020 and is based in the Netherlands. They
+have a relatively small team but seem to have a fast growing user base,
+and very enthusiastic developers. Polars is written in Rust and
+available for Python. Because it is written in Rust (which is a low
+level language), it is very fast, and can handle all common data
+formats. The two main developers are described as enthusiastic about
+open source.
+
+::: columns
+::: column
+#### Benefits
+
+-   Enthusiastic user community, with a lot of support available.
+-   Supports several languages, including Python.
+-   Handles all common data formats.
+-   Faster than Pandas.
+-   Polars have an extensive user guide section.
+:::
+
+::: column
+#### Drawbacks
+
+-   Relatively young project, so may not have all the features of more
+    established tools.
+-   Because it is still new, it isn't as widely used and doesn't have
+    too many learning resources available yet.
+-   No traditional documentation section, everything is in the user
+    guide.
```

Comment:
I'm not sure I understand what you mean here by "traditional documentation". Can you expand?

Reply:
There isn't a header on the website saying Documentation, you have to dig into the User Guides to find it. I may just have been lucky, but I feel that almost all the tools I've had a look at have had a section called something like Documentation.